src/Adapters/ADTAdapter.ts

```diff
@@ -472,6 +474,26 @@ export default class ADTAdapter implements IADTAdapter {
         );
     }
 
+    searchTwinsByQuery(params: AdapterMethodParamsForSearchTwinsByQuery) {
+        const adapterMethodSandbox = new AdapterMethodSandbox(this.authService);
+        return adapterMethodSandbox.safelyFetchDataCancellableAxiosPromise(
```

Comment:
I'm still expecting to see the recursion to fetch all the pages of results in this PR since it's primarily for the querying logic.